rholang/src/main/scala/coop/rchain/rholang/interpreter/Substitute.scala

```diff
@@ -55,7 +55,8 @@ object Substitute {
       Nil,
       for { m <- term.matches } yield substitute(m),
       term.id,
-      term.freeCount
+      term.freeCount,
+      term.locallyFree.from(env.level).map(x => x - env.level)
```

Comment:
Why is env.level defined as

`def level: Int = env.keys.max + 1`

and not just

`def level: Int = env.keys.max`

Reply:
Well I guess the definition of env.level is fine but don't we have to do

`term.locallyFree.from(env.level - 1).map(x => x - env.level + 1)`

Reply:
No. If the max key in the env is 0, then we need to shift the environment by 1.